### PR TITLE
Cas branch

### PR DESCRIPTION
**Pull request by Casvharen**
This PR makes sure that
1. updating rounds works because the PUT is now supported by a route that updates the rounds status
2. Results can be retrieved for the end of a round
3. The round ready attribute is reused for when users submit all their answers and is then checked to set the round to finished.
4. minor stuff I can't remember...

### CODE REVIEW

File: app/Http/Controllers/GuessController.php [hunk 1]
```diff
@@ -28,11 +28,19 @@ public function store(Request $request): JsonResponse {
...
+        if($guess === null){
+            $guess = Guess::create([
+                'track_id' => $track_id,
+                'user_id' => $request->get('user_id'),
+                'guessed_user_id' => $request->get('guessed_user_id'),
+            ]);
+        } else {
+            $guess->guessed_user_id = $request->get('guessed_user_id');
+            $guess->save();
+        }
```
Comment (Tomgroot) [inline app/Http/Controllers/GuessController.php:42]: Would prefer keeping PUT and POST calls separately. PUT for updating and POST for creating. For validation when editing, use rules() in the entity and $this->rules for creation.

File: app/Models/Round.php [hunk 2]
```diff
@@ -53,12 +53,56 @@ public function getCreatorAttribute() {
...
+        if ($this->tracks()->count() <= 2 || $this->tracks()->pluck('ready')->contains(0)) {
             return;
         }
 
-        $this->status = self::STATUS_GUESS_WHOSE;
+        // TODO: there must be a better way to do this...
+        foreach($this->tracks as $track){
+            $track->ready = 0;
+            $track->save();
+        };
```
Comment (Tomgroot) [inline app/Models/Round.php:65]: What do we want to accomplish here?

File: app/Models/Round.php [hunk 3]
```diff
@@ -53,12 +53,56 @@ public function getCreatorAttribute() {
...
+        foreach($this->tracks as $track){
+
+            foreach($track->guesses->sortBy('user_id')->values() as $key => $guess){
+                $extra = (int)($guess->guessed_user_id == $track->user_id || $users[$key]->user_id == $track->user_id);
+                $users[$key]->score += $extra;
+            }
+        }
+
+        // Not super happy with this, but otherwise a foreach counter is needed because keys are messed up
+        $users = array_values($users->sortBy('score', SORT_REGULAR, true)->values()->all());
```
Comment (Tomgroot) [inline app/Models/Round.php:93]: The keys are the users ids or? Let me check this tomorrow

File: app/Models/Round.php (hunk 2 above)
Comment (Casvharen) [reply app/Models/Round.php:65]: The ready status can be reused for checking if everyone filled in their guess and is happy with them, so setting them back to 0 when the round is in a new state seems usefull.

File: app/Models/Round.php (hunk 3 above)
Comment (Casvharen) [reply app/Models/Round.php:93]: They don't seem to be

File: app/Models/Round.php (hunk 2 above)
Comment (Tomgroot) [reply app/Models/Round.php:67]: Track->ready depicts if the track is ready to be guessed right? Why should we want to revert this at any time? A new round gets new tracks.

**Pull request merged**